packages/purpose-process/test/testDeletePurpose.ts

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { describe, expect, it } from "vitest";
+import {
+  DraftPurposeDeletedV2,
+  Purpose,
+  PurposeId,
+  PurposeVersion,
+  WaitingForApprovalPurposeDeletedV2,
+  generateId,
+  purposeVersionState,
+  toPurposeV2,
+  toReadModelEService,
+} from "pagopa-interop-models";
+import {
+  getMockPurpose,
+  writeInReadmodel,
+  readLastEventByStreamId,
+  decodeProtobufPayload,
+  getMockPurposeVersion,
+} from "pagopa-interop-commons-test";
+import {
+  purposeNotFound,
+  organizationIsNotTheConsumer,
+  purposeCannotBeDeleted,
+} from "../src/model/domain/errors.js";
+import {
+  eservices,
+  postgresDB,
+  purposeService,
+  purposes,
+} from "./purposeService.integration.test.js";
+import { addOnePurpose, getMockEService } from "./utils.js";
+
+export const testDeletePurpose = (): ReturnType<typeof describe> =>
+  describe("deletePurpose", () => {
+    it("should write on event-store for the deletion of a purpose (no versions)", async () => {
```

Comment:
```suggestion
    it.each([
      { description: "no versions", versions: [] },
      {
        description: "draft version",
        versions: [getMockPurposeVersion(purposeVersionState.draft)],
      },
      {
        description: "waiting for approval version",
        versions: [
          getMockPurposeVersion(purposeVersionState.waitingForApproval),
        ],
      },
    ])(
      "should write on event-store for the deletion of a purpose with $description",
      async ({ versions }) => {
```
With this you can cover the first three tests